We need insights, not data (#281)
mauroservienti/milestone.github.io: _posts/we-need-insights-not-data.md

---
layout: post
title: We need insights, not data
author: Mauro Servienti
title: "Gauges and graphs attract software engineers like honey for bees. We spend hours implementing distributed logging solutions or monitoring systems, and still we have a hard time understanding what's going on."
header_image: /img/posts/we-need-insights-not-data/header.jpg
tags:
- personal
- distributed-systems
---

I used a [Whoop](https://www.whoop.com) device for about 18 months. What's Whoop, you might wonder?

Quoting from their website:

> The latest, most advanced fitness and health wearable available. Monitor your recovery, sleep, training, and health, with personalized recommendations and coaching feedback.

> I stopped using it because I got no personalized recommendations and coaching feedback. And if you recall, it's the thing I talked about at the [end of 2021 in "Do I still need a smartwatch?"](https://milestone.topics.it/2021/12/27/do-I-still-need-a-smartwatch.html):

Reviewer: I suggest removing references to readers who follow regularly. The post may get more hits from a Google search or some other random visitor who may not recall this at all.

```suggestion
I stopped using it because I got no personalized recommendations and coaching feedback. I mentioned this at the [end of 2021 in "Do I still need a smartwatch?"](https://milestone.topics.it/2021/12/27/do-I-still-need-a-smartwatch.html):

```

Garmin's training status feature, instead of simply reporting your current status, comes with suggestions on how to change your training routine to be more productive. My tips were to either train more often or, better yet, change the training sessions to be more interval-style and more challenging for my cardiovascular system. For example, instead of bike riding for 80 kilometers twice a week, do four sessions, three of which could be shorter and focused on aerobic and high aerobics intensity.

The great thing is that the device even comes with suggestions for today's session. It uses the body recovery status to determine what's best, including resting. So, today's advice might be to do nothing and focus on relaxing. Interestingly, the Garmin algorithm uses the same raw data Whoop captures. The Fenix device has a heartbeat and pulse oximeter sensor similar to the Whoop one. It tracks more or less the same raw data, such as resting heartbeats, respiration, sleep, heart rate variability, etc. 

> In my specific case, what's reassuring is that Gaming insights are well aligned with the topics I recently discussed with my friend [Daniel](https://twitter.com/danielmarbach). Daniel is not only a brilliant software engineer, but he's also a sportsman that knows a lot about training. I presented him with my issues and gut feelings, and he gave me the same advice Garmin gave me a couple of weeks later.

Reviewer: ```suggestion
In my specific case, what's reassuring is that Garmin insights are well aligned with the topics I recently discussed with my friend [Daniel](https://twitter.com/danielmarbach). Daniel is not only a brilliant software engineer, but he's also a sportsman that knows a lot about training. I presented him with my issues and gut feelings, and he gave me the same advice Garmin gave me a couple of weeks later.

```

## What about software systems?

> For a very long time, I had this gut feeling that monitoring dashboards were useless, if not worse. All those gauges and graphs attract us—bells and whistles of the digital era. If a monitoring dashboard provides only raw data, we need an external coach to understand what actions we need to move forward and improve.

Reviewer: ```suggestion
For a very long time, I had this gut feeling that monitoring dashboards were useless at best. All those gauges and graphs attract us—bells and whistles of the digital era. If a monitoring dashboard provides only raw data, we need an external coach to understand what actions we need to move forward and improve.

```
